src/OpenSage.Game/Network/EchoConnection.cs

```diff
@@ -1,42 +1,63 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
+using NLog;
 using OpenSage.Logic.Orders;
+using OpenSage.Network.Packets;
 
 namespace OpenSage.Network
 {
-    public sealed class EchoConnection : IConnection
+    public class EchoConnection : IConnection
     {
-        private struct ReceivedPacket
-        {
-            public uint Frame;
-            public List<Order> Orders;
-        }
+        private static readonly NLog.Logger Logger = NLog.LogManager.GetCurrentClassLogger();
 
-        private readonly List<ReceivedPacket> _receivedPackets = new List<ReceivedPacket>();
+        private readonly List<SkirmishOrderPacket> _receivedPackets = new List<SkirmishOrderPacket>();
 
-        public void Send(uint frame, List<Order> orders)
+        public virtual void Send(uint frame, List<Order> orders)
         {
-            _receivedPackets.Add(new ReceivedPacket
+            StorePacket(new SkirmishOrderPacket
             {
                 Frame = frame,
-                Orders = orders
+                Orders = orders.ToArray()
```

Comment:
This causes an array allocation (clone + copy) for every single sent packet. Probably not an issue at this point, but if we want to avoid frequent allocations as much as possible then we should figure out something to avoid this (later).